drivers/vulkan/rendering_device_driver_vulkan.cpp

```diff
@@ -0,0 +1,3247 @@
+	image_view_create_info.subresourceRange.levelCount = create_info.mipLevels;
+	image_view_create_info.subresourceRange.layerCount = create_info.arrayLayers;
+	if ((p_format.usage_bits & TEXTURE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT)) {
+		image_view_create_info.subresourceRange.aspectMask = VK_IMAGE_ASPECT_DEPTH_BIT;
+	} else {
+		image_view_create_info.subresourceRange.aspectMask = VK_IMAGE_ASPECT_COLOR_BIT;
+	}
+
+	VkImageView vk_image_view = VK_NULL_HANDLE;
+	err = vkCreateImageView(vk_device, &image_view_create_info, nullptr, &vk_image_view);
+	if (err) {
+		vmaDestroyImage(allocator, vk_image, allocation);
+		ERR_FAIL_COND_V_MSG(err, TextureID(), "vkCreateImageView failed with error " + itos(err) + ".");
+	}
+
+	// Bookkeep.
+
+	TextureInfo ti;
+	ti.vk_image = vk_image;
+	ti.owner_info.allocation.handle = allocation;
+	ti.owner_info.allocation.size = alloc_info.size;
+	ti.owner_info.vk_create_info.format = image_view_create_info.format;
+	ti.owner_info.vk_create_info.level_count = create_info.mipLevels;
+	ti.owner_info.vk_create_info.layer_count = create_info.arrayLayers;
+	ti.owner_info.vk_create_info.usage_flags = create_info.usage;
+	ti.owner_info.vk_create_info.view_type = image_view_create_info.viewType;
+	ti.owner_info.vk_create_info.aspect_mask = image_view_create_info.subresourceRange.aspectMask;
+
+	textures.insert(vk_image_view, ti);
+
+	return TextureID(vk_image_view);
+}
+
+RDD::TextureID RenderingDeviceDriverVulkan::texture_create_from_extension(uint64_t p_native_texture, TextureType p_type, DataFormat p_format, uint32_t p_array_layers, bool p_depth_stencil) {
+	VkImage vk_image = (VkImage)p_native_texture;
+
+	// We only need to create a view into the already existing natively-provided texture.
+
+	VkImageViewCreateInfo image_view_create_info = {};
+	image_view_create_info.sType = VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO;
+	image_view_create_info.image = vk_image;
+	image_view_create_info.viewType = (VkImageViewType)p_type;
+	image_view_create_info.format = RD_TO_VK_FORMAT[p_format];
+	image_view_create_info.components.r = VK_COMPONENT_SWIZZLE_R;
+	image_view_create_info.components.g = VK_COMPONENT_SWIZZLE_G;
+	image_view_create_info.components.b = VK_COMPONENT_SWIZZLE_B;
+	image_view_create_info.components.a = VK_COMPONENT_SWIZZLE_A;
+	image_view_create_info.subresourceRange.levelCount = 1;
+	image_view_create_info.subresourceRange.layerCount = p_array_layers;
+	image_view_create_info.subresourceRange.aspectMask = p_depth_stencil ? VK_IMAGE_ASPECT_DEPTH_BIT : VK_IMAGE_ASPECT_COLOR_BIT;
+
+	VkImageView vk_image_view = VK_NULL_HANDLE;
+	VkResult err = vkCreateImageView(vk_device, &image_view_create_info, nullptr, &vk_image_view);
+	if (err) {
+		ERR_FAIL_COND_V_MSG(err, TextureID(), "vkCreateImageView failed with error " + itos(err) + ".");
+	}
+
+	// Bookkeep.
+
+	TextureInfo ti;
+	ti.vk_image = vk_image;
+
+	textures.insert(vk_image_view, ti);
+
+	return TextureID(vk_image);
```

Comment:
This needs to be `return TextureID(vk_image_view);` just like the other create functions.

Reply:
Done! Will come in the next push.